### PR TITLE
appearance.ui (cursor size scale)

### LINKED ISSUE #304: Pointer slider is not updatable with left/right arrows [15$]
#### Expected behaviour
The pointer slider should be editable with left/right arrows as it is possible with other slider (sound) on the Mate desktop.

#### Actual behaviour
The slider to increase/decrease the pointer size is not modifiable with the keyboard. Using left/right produces no effect.  Origin/end works correctly.

#### Steps to reproduce the behaviour
1) Open appearance
2) Open personalize 
3) Go to the pointer tab
4) Move the keyboard cursor to the pointer size slider with tab
5) Press right or left arrow

#### MATE general version
1.8 on Debian Jessie and 1.18 on Ubuntu 17.10

#### Linux Distribution
Both Ubuntu 17.10 and Debian 8.9.

Best regards.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/48917885-pointer-slider-is-not-updatable-with-left-right-arrows?utm_campaign=plugin&utm_content=tracker%2F559197&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F559197&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by tamplan**
Add Adjudtement to change value by left/right arrows

Fix #304

### CODE REVIEW

File: capplets/desktop-icons/mate-desktop-icons-properties.ui~ [hunk 1]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rbuj) [inline capplets/desktop-icons/mate-desktop-icons-properties.ui~]: It looks a lot like a backup file.

Comment (tamplan) [reply capplets/desktop-icons/mate-desktop-icons-properties.ui~]: Yes it is, sorry...

File: capplets/desktop-icons/mate-desktop-icons.pc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+prefix=/usr/local
```
Comment (rbuj) [inline capplets/desktop-icons/mate-desktop-icons.pc]: You can also remove the file mate-desktop-icons.pc.

Comment (tamplan) [reply capplets/desktop-icons/mate-desktop-icons.pc]: Done.

**Comment by tamplan** [comment]
> Please rebase against master.

I think it's done.

File: capplets/desktop-icons/mate-di-capplet.h [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ *  as published by the Free Software Foundation
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Street #330, Boston, MA 02110-1301, USA.
```
Comment (rbuj) [inline capplets/desktop-icons/mate-di-capplet.h]: Update [FSF address](https://www.fsf.org/about/contact/): 
```
Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```

File: capplets/desktop-icons/mate-di-capplet.c [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+ *  as published by the Free Software Foundation
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Street #330, Boston, MA 02110-1301, USA.
```
Comment (rbuj) [inline capplets/desktop-icons/mate-di-capplet.c]: Update [FSF address](https://www.fsf.org/about/contact/): 
```
Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```

Comment (tamplan) [reply capplets/desktop-icons/mate-di-capplet.c]: Copy/pate from https://github.com/tamplan/mate-control-center/blob/f741389053ce946aca1baa227c6f2fb011469d72/capplets/default-applications/mate-da-capplet.c#L19

Done in my files, preparing new PR for 4/5 files with wrong FSF address...

File: capplets/desktop-icons/mate-di-capplet.c [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+	capplet->show_icons_settings = g_settings_new (SHOW_ICONS_SCHEMA);
+	capplet->desktop_icons_settings = g_settings_new (DESKTOP_ICONS_SCHEMA);
+
+	capplet->button_show_icons_toggle = get_widget("button_show_icons_toggle");
+	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
+
+	/* CheckButton to allow icon showing or not */
+	gtk_toggle_button_set_active(capplet->button_show_icons_toggle, g_settings_get_boolean(capplet->show_icons_settings, SHOW_ICONS_KEY));
+
+	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
```
Comment (rbuj) [inline capplets/desktop-icons/mate-di-capplet.c]: **cppcheck warning**
Id: redundantAssignment (see line 294)
Variable 'capplet->desktop_icons_caption' is reassigned a value before the old one has been used.

File: capplets/desktop-icons/mate-di-capplet.c [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+{
+	/* Get computer icon name if set by the user */
+	g_autoptr(GVariant) *user_computer_name = g_settings_get_user_value(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+	if (user_computer_name != NULL)
+	{
+		gchar *computer_name = g_settings_get_string(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+		gtk_button_set_label(GTK_BUTTON(capplet->button_computer_toggle), computer_name);
+		g_free(computer_name);
+	}
+	g_variant_unref(user_computer_name);
```
Comment (rbuj) [inline capplets/desktop-icons/mate-di-capplet.c]: `g_variant_unref` can't be used with `g_autoptr(GVariant)`. Remove:
```C
g_variant_unref(user_computer_name);
g_variant_unref(user_home_name);
g_variant_unref(user_trash_name);
g_variant_unref(user_network_name);
```

File: capplets/desktop-icons/mate-di-capplet.c [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+
+	/* Get home icon name if set by the user */
+	g_autoptr(GVariant) *user_home_name = g_settings_get_user_value(capplet->desktop_icons_settings, HOME_NAME_KEY);
+	if (user_home_name != NULL)
+	{
+		gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);
```
Comment (rbuj) [inline capplets/desktop-icons/mate-di-capplet.c]: You can use `g_autofree gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);`, then you shoud remove `g_free(home_name);`

**Comment by rbuj** [comment]
```patch
--- a/capplets/desktop-icons/mate-di-capplet.c
+++ b/capplets/desktop-icons/mate-di-capplet.c
@@ -47,7 +47,7 @@ close_cb(GtkWidget* window, gint response, MateDICapplet* capplet)
 static void
 show_icons_cb(GtkCheckButton* button, MateDICapplet* capplet)
 {
-	gboolean show_icons = gtk_toggle_button_get_active(button);
+	gboolean show_icons = gtk_toggle_button_get_active(GTK_BUTTON(button));
 	
 	gtk_widget_set_sensitive(capplet->desktop_icons_caption, show_icons);
 
@@ -65,10 +65,10 @@ static void
 edit_computer_icon_name_cb(GtkWidget* button, MateDICapplet* capplet)
 {
 	/* We enter in edit mode so change GUI to enter new name */
-	if (strcmp( gtk_button_get_label(button), "gtk-edit" ) == 0 )
+	if (strcmp( gtk_button_get_label(GTK_BUTTON(button)), "gtk-edit" ) == 0 )
 	{
 		gtk_widget_hide(capplet->button_computer_toggle);
-		gtk_entry_set_text(capplet->entry_computer, gtk_button_get_label(capplet->button_computer_toggle) );
+		gtk_entry_set_text(GTK_ENTRY(capplet->entry_computer), gtk_button_get_label(GTK_BUTTON(capplet->button_computer_toggle)) );
 		gtk_widget_show(capplet->entry_computer);
 		gtk_button_set_label(GTK_BUTTON(button), "gtk-undo");
 	}
@@ -104,10 +104,10 @@ static void
 edit_home_icon_name_cb(GtkWidget* button, MateDICapplet* capplet)
 {
 	/* We enter in edit mode so change GUI to enter new name */
-	if (strcmp( gtk_button_get_label(button), "gtk-edit" ) == 0 )
+	if (strcmp( gtk_button_get_label(GTK_BUTTON(button)), "gtk-edit" ) == 0 )
 	{
 		gtk_widget_hide(capplet->button_home_toggle);
-		gtk_entry_set_text(capplet->entry_home, gtk_button_get_label(capplet->button_home_toggle) );
+		gtk_entry_set_text(GTK_ENTRY(capplet->entry_home), gtk_button_get_label(GTK_BUTTON(capplet->button_home_toggle)) );
 		gtk_widget_show(capplet->entry_home);
 		gtk_button_set_label(GTK_BUTTON(button), "gtk-undo");
 	}
@@ -143,10 +143,10 @@ static void
 edit_trash_icon_name_cb(GtkWidget* button, MateDICapplet* capplet)
 {
 	/* We enter in edit mode so change GUI to enter new name */
-	if (strcmp( gtk_button_get_label(button), "gtk-edit" ) == 0 )
+	if (strcmp( gtk_button_get_label(GTK_BUTTON(button)), "gtk-edit" ) == 0 )
 	{
 		gtk_widget_hide(capplet->button_trash_toggle);
-		gtk_entry_set_text(capplet->entry_trash, gtk_button_get_label(capplet->button_trash_toggle) );
+		gtk_entry_set_text(GTK_ENTRY(capplet->entry_trash), gtk_button_get_label(GTK_BUTTON(capplet->button_trash_toggle)) );
 		gtk_widget_show(capplet->entry_trash);
 		gtk_button_set_label(GTK_BUTTON(button), "gtk-undo");
 	}
@@ -182,10 +182,10 @@ static void
 edit_network_icon_name_cb(GtkWidget* button, MateDICapplet* capplet)
 {
 	/* We enter in edit mode so change GUI to enter new name */
-	if (strcmp( gtk_button_get_label(button), "gtk-edit" ) == 0 )
+	if (strcmp( gtk_button_get_label(GTK_BUTTON(button)), "gtk-edit" ) == 0 )
 	{
 		gtk_widget_hide(capplet->button_network_toggle);
-		gtk_entry_set_text(capplet->entry_network, gtk_button_get_label(capplet->button_network_toggle) );
+		gtk_entry_set_text(GTK_ENTRY(capplet->entry_network), gtk_button_get_label(GTK_BUTTON(capplet->button_network_toggle)) );
 		gtk_widget_show(capplet->entry_network);
 		gtk_button_set_label(GTK_BUTTON(button), "gtk-undo");
 	}
@@ -221,44 +221,36 @@ void
 get_user_icons_names(MateDICapplet* capplet)
 {
 	/* Get computer icon name if set by the user */
-	g_autoptr(GVariant) *user_computer_name = g_settings_get_user_value(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+	g_autoptr(GVariant) user_computer_name = g_settings_get_user_value(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
 	if (user_computer_name != NULL)
 	{
-		gchar *computer_name = g_settings_get_string(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
+		g_autofree gchar *computer_name = g_settings_get_string(capplet->desktop_icons_settings, COMPUTER_NAME_KEY);
 		gtk_button_set_label(GTK_BUTTON(capplet->button_computer_toggle), computer_name);
-		g_free(computer_name);
 	}
-	g_variant_unref(user_computer_name);
 
 	/* Get home icon name if set by the user */
-	g_autoptr(GVariant) *user_home_name = g_settings_get_user_value(capplet->desktop_icons_settings, HOME_NAME_KEY);
+	g_autoptr(GVariant) user_home_name = g_settings_get_user_value(capplet->desktop_icons_settings, HOME_NAME_KEY);
 	if (user_home_name != NULL)
 	{
-		gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);
+		g_autofree gchar *home_name = g_settings_get_string(capplet->desktop_icons_settings, HOME_NAME_KEY);
 		gtk_button_set_label(GTK_BUTTON(capplet->button_home_toggle), home_name);
-		g_free(home_name);
 	}
-	g_variant_unref(user_home_name);
 
 	/* Get trash icon name if set by the user */
-	g_autoptr(GVariant) *user_trash_name = g_settings_get_user_value(capplet->desktop_icons_settings, TRASH_NAME_KEY);
+	g_autoptr(GVariant) user_trash_name = g_settings_get_user_value(capplet->desktop_icons_settings, TRASH_NAME_KEY);
 	if (user_trash_name != NULL)
 	{
-		gchar *trash_name = g_settings_get_string(capplet->desktop_icons_settings, TRASH_NAME_KEY);
+		g_autofree gchar *trash_name = g_settings_get_string(capplet->desktop_icons_settings, TRASH_NAME_KEY);
 		gtk_button_set_label(GTK_BUTTON(capplet->button_trash_toggle), trash_name);
-		g_free(trash_name);
 	}
-	g_variant_unref(user_trash_name);
 
 	/* Get network icon name if set by the user */
-	g_autoptr(GVariant) *user_network_name = g_settings_get_user_value(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
+	g_autoptr(GVariant) user_network_name = g_settings_get_user_value(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
 	if (user_network_name != NULL)
 	{
-		gchar *network_name = g_settings_get_string(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
+		g_autofree gchar *network_name = g_settings_get_string(capplet->desktop_icons_settings, NETWORK_NAME_KEY);
 		gtk_button_set_label(GTK_BUTTON(capplet->button_network_toggle), network_name);
-		g_free(network_name);
 	}
-	g_variant_unref(user_network_name);
 }
 
 /* Init GUI reading user preferences if set */
@@ -294,9 +286,7 @@ init_dialog(MateDICapplet* capplet)
 	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
 
 	/* CheckButton to allow icon showing or not */
-	gtk_toggle_button_set_active(capplet->button_show_icons_toggle, g_settings_get_boolean(capplet->show_icons_settings, SHOW_ICONS_KEY));
-
-	capplet->desktop_icons_caption = get_widget("desktop_icons_caption");
+	gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(capplet->button_show_icons_toggle), g_settings_get_boolean(capplet->show_icons_settings, SHOW_ICONS_KEY));
 
 	/* One CheckButton for each icon to show or hide*/
 	capplet->button_computer_toggle = get_widget("button_computer_toggle");
```

**Comment by tamplan** [comment]
Thank for you help.

I realize i have made a mistake merging the desktop_icons branch (to develop new applet) in master branch with a PR awaiting validation to resolve an issue...

Sorry for that!

**Comment by tamplan** [comment]
I need to review more my code for the applet and perhaps learn more the use of git too...

**Comment by tamplan** [comment]
I need to review more my code for the applet and perhaps learn more the use of git too...
To many commits for too subjects, PR to resolve issue and PR to add new feature.
Prepare 2 PR, one by subject.

**Pull request closed without merging**